### PR TITLE
Correct prescaler calculation for BLHeli_S

### PR DESCRIPTION
**Pull request by andyp1per**
Experimentation shows that BLHeli_S wants a prescaler that gives at least the requested clock frequency. This is particularly true for BlueJay ESCs. Our old (incorrect) algorithm was actually doing this, but now that the algorithm has been corrected it is no longer working.

### CODE REVIEW

**Comment by Hwurzburg** [comment]
how does this error impact users? what versions of firmware? what conditions?

**Comment by andyp1per** [comment]
It's only a bug in master - currently in master BLHeli_S motors will not spin due to this issue

**Comment by tridge** [comment]
@andyp1per does this explain the de-syncs you've seen?

**Comment by andyp1per** [comment]
@tridge - no, this only affects 4.5 and I have been flying 4.4

**Comment by andyp1per** [comment]
also my Chimera has BLHeli32 which is unchanged

File: libraries/AP_HAL/RCOutput.cpp [hunk 1]
```diff
@@ -76,7 +76,7 @@ bool AP_HAL::RCOutput::is_dshot_protocol(const enum output_mode mode)
 /*
  * calculate the prescaler required to achieve the desire bitrate
  */
-uint32_t AP_HAL::RCOutput::calculate_bitrate_prescaler(uint32_t timer_clock, uint32_t target_frequency, bool is_dshot)
+uint32_t AP_HAL::RCOutput::calculate_bitrate_prescaler(uint32_t timer_clock, uint32_t target_frequency, bool at_least_freq)
```
Comment (tridge) [inline libraries/AP_HAL/RCOutput.cpp:79]: the callers pass in an is_dshot. The test suite sometimes passes true and sometimes false for this variable. So the users of this function don't seem to match the test suite or the new function form?

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by tridge** [comment]
if this is supposed to be based on the ESC type parameter then I don't see how this parameter gets into the function that is changing

**Review by tridge — APPROVED** [review]

**Pull request merged**